synapse/handlers/room_member.py

```diff
@@ -213,11 +213,11 @@ def _local_membership_update(
 
             if predecessor:
                 # It is an upgraded room. Copy over old tags
-                self.copy_room_tags_and_direct_to_room(
+                yield self.copy_room_tags_and_direct_to_room(
```

Comment:
it's great that you're finding and fixing these, though I strongly recommend sticking them in a separate PR